Automate the process of creating new releases #92

Ph0enixKM opened the pull request.

Ph0enixKM commented:
After some struggles trying to configure this on my own I gave up and got this working with `cargo-dist`. This automation pipeline works for:
- Linux x86_64
- macOS x86_64
- macOS ARM

Unfortunately the `cargo-dist` **does not support Linux ARM**. That's the only downside.

The installer script has been adjusted to work for the new github actions pipeline. Let me know if you think we can do it better someway.

To test this:
1. Fork this repo or import it and merge this branch into master
2. Update release in `Cargo.toml` to anything (ex. `1.1.1`)
3. Create a new release with tag that matches the `Cargo.toml` release.
4. Wait for the Github Action to finish
5. Uninstall amber if it's installed already
6. Run the installer script with updated `url` variable in `install.ab` to match your repo url (`cargo run setup/install.ab`) 
7. Done ✨

arapower reviewed (approved):
LGTM.
Let's improve Linux ARM support later.

Ph0enixKM commented:
I'll try to compile linux arm in a separate github workflow

Ph0enixKM commented:
I managed to do this. You can test it one more time if you'd like

arapower commented on .github/workflows/release-linux-arm.yml:
```diff
@@ -0,0 +1,47 @@
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
```suggestion
        uses: actions/checkout@v4
```

arapower commented on .github/workflows/release-linux-arm.yml:
```diff
@@ -0,0 +1,47 @@
+          tar -cJf amber-aarch64-unknown-linux-gnu.tar.xz amber
+
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
```
```suggestion
        uses: actions/upload-artifact@v4
```

arapower commented on .github/workflows/release-linux-arm.yml:
```diff
@@ -0,0 +1,47 @@
+
+      - name: Upload Release Asset
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1
```
`actions/upload-release-asset` is archived.
How about using https://github.com/softprops/action-gh-release instead?

Ph0enixKM commented:
@arapower these version changes won't introduce any breaking changes?

Ph0enixKM commented:
@arapower All issues are now resolved:
- Upgraded `actions/checkout` version
- Replaced deprecated `actions/upload-release-asset` with `softprops/action-gh-release`
- Removed unnecessary workflow `actions/upload-artifact`

Ph0enixKM commented:
This issue introduces a version bump to `0.3.2-alpha`

arapower commented:
@Ph0enixKM 
I don't have ARM Linux, so I can't verify if the process works properly.

Ph0enixKM commented:
@arapower if the ARM binary doesn't work for someone then we can adjust it later. It should though as the workflow is using `cross-rs` to compile it and it doesn't fail.

arapower reviewed (approved):
LGTM
